design-proposals/eim-standalone-managed-en.md

```diff
@@ -0,0 +1,236 @@
+5. The customer can now provision additional Edge Nodes via remote provisioning or manually create additional SENs and follow the same workflow to onboard them.
+
+This document describes the design of the onboarding process of Standalone Edge Nodes to managed Edge Nodes (step 3) to allow for all further steps. 
+
+> NOTE: Converting standalone K8s clusters and applications to managed clusters and applications is out of scope for this document. 
+
+## Proposal
+
+### Design goals
+
+This design aims at:
+
+- Providing a solution for onboarding Edge Nodes in a fully automated way, with minimal manual steps.
+- Keeping the solution OS-independent, i.e., the solution should work on any OS that is supported by the Edge Orchestrator.
+- Reducing the user interaction (i.e., logging into the OS, injecting USB sticks, etc.) with Edge Node machines.
+- Enabling the onboarding process at scale, i.e., the design should enable onboarding of multiple machines at once.
+
+The proposed solution is an MVP approach - some of the above goals may not be fully achieved in the first version of the design,
+but the design doesn't preclude achieving them in the future. 
+
+### Assumptions
+
+- Customers will drive the onboarding process from a local developer machine, with desktop, keyboard and mouse.
+- The local developer machine will have direct access to ENs via local subnet. Customers can SSH into the ENs.
+- SENs being onboarded won't be equipped with desktop, keyboard or mouse. The only machine equipped with peripherals will be
+  the local developer/admin machine.
+- The design will re-use the current APIs of Onboarding Manager to drive IO/NIO-based onboarding.
+  We may just require small modifications to the Onboarding Manager and the gRPC interface.
+
+### Solution
+
+```mermaid
+sequenceDiagram
+%%{wrap}%%
+autonumber
+
+  box LightYellow Edge Node(s)
+  participant ctl as Standalone Onboarding CLI
+  end
+
+  participant user as User
+
+  box rgb(235,255,255) Edge Orchestrator
+
+  participant pa as Provisioning Nginx
+  participant om as Onboarding Manager
+  participant kc as KeyCloak
+  participant inv as Inventory / API
+
+  end
+
+  note over ctl,user: User installs Standalone EN, decides to scale out
+
+  note over pa,inv: OS profiles created beforehand, no Host/Instance pre-registration
+
+  user->>user: Retrieves standalone onboarding user credentials 
+  user->>ctl: Logs in to the EN and invokes CLI tool
+  
+  note over ctl: CLI triggered
+
+  ctl->>ctl: Checks local OS prerequisites
+
+  ctl->>ctl: Take user inputs and validate
+
+  ctl->>+pa: Get orchestrator CA certificate
+  pa-->>-ctl: [CA certificate]
+
+  ctl->>+kc: Retrieve JWT token for standalone onboarding user
+  kc->>-ctl: [JWT token]
+
+  ctl->>+om: Invoke OnboardStandaloneService gRPC (UUID, Serial Number, OS info, ...)
+
+  om->>om: Validate JWT standalone onboarding role
+
+  om->>inv: Query Inventory for OS resource by OS version and distribution
```

Comment:
what if the OS version does not match the OS resource ?

Reply:
we should just return an error to the caller - so in generl we can always retry